File: solutions/2026-07-21/solution.java
// Container With Most Water
// Difficulty: Hard
// URL: https://leetcode.com/problems/container-with-most-water/

public class Solution {
    public int[] findMidpoint(int[] nums) {
        if (nums.length == 0 || nums.length == 1) {
            return null;
        }
        int target = (nums.length + 1) / 2;
        int left = 0, right = nums.length - 1;
        while (left <= right) {
            int currentSum = nums[left] + nums[right];
            if (currentSum == target) {
                return new int[] {left + 1, right + 1};
            } else if (currentSum < target) {
                left++;
            } else {
                right--;
            }
        }
        return null;
    }
}

File: solutions/2026-07-21/solution.py
# Container With Most Water
# Difficulty: Hard
# URL: https://leetcode.com/problems/container-with-most-water/

def findMidpoint(nums, n):
    if n == 0 or n == 1:
        return None
    target = (n + 1) // 2
    left, right = 0, n - 1
    while left <= right:
        current_sum = nums[left] + nums[right]
        if current_sum == target:
            return [left + 1, right + 1]
        elif current_sum < target:
            left += 1
        else:
            right -= 1
    return None